contentcuration/contentcuration/management/commands/syncnodes.py

```diff
@@ -0,0 +1,110 @@
+import copy
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from contentcuration.models import Channel, ContentNode, AssessmentItem, File, ContentTag
+from le_utils.constants import content_kinds
+
+import logging as logmodule
+logmodule.basicConfig()
+logging = logmodule.getLogger(__name__)
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('channel_id', type=str)
+        parser.add_argument('--sort', action='store_true', dest='sort', default=False)
+        parser.add_argument('--tags', action='store_true', dest='tags', default=False)
+        parser.add_argument('--files', action='store_true', dest='files', default=False)
+        parser.add_argument('--assessment-items', action='store_true', dest='assessment_items', default=False)
+
+    def handle(self, *args, **options):
+        channel = Channel.objects.get(pk=options['channel_id'])
+        sync_sort_order = options['sort']
+        sync_tags = options['tags']
+        sync_files = options['files']
+        sync_assessment_items = options['assessment_items']
+        parents_to_check = [] # Keep track of parents to make resorting easier
+
+        with transaction.atomic():
+            with ContentNode.objects.delay_mptt_updates():
+                logging.info("Syncing nodes for channel {} (id:{})".format(channel.name, channel.pk))
+                for node in channel.main_tree.get_descendants():
+                    original_node = node.get_original_node()
+                    if original_node.node_id != node.node_id: # Only update if node is not original
+                        sync_node(node, original_node, sync_order=sync_sort_order)
+                        if sync_sort_order and node.parent not in parents_to_check:
+                            parents_to_check.append(node.parent)
+                        if sync_tags:
+                            sync_node_tags(node, original_node, options['channel_id'])
+                        if sync_files:
+                            sync_node_files(node, original_node)
+                        if sync_assessment_items and node.kind_id == content_kinds.EXERCISE:
+                            sync_node_assessment_items(node, original_node)
+
+                # Avoid cases where sort order might have overlapped
+                for parent in parents_to_check:
```

Comment:
Likely for same sort_order to be set if a user adds own content to same level- ensure sort orders are resolved